NuMPI/Optimization/LBFGS.py

```diff
@@ -406,16 +415,38 @@ def _phi_phiprime(alpha):
             **linesearch_options)
 
         printdb("derphi: {}".format(derphi))
-        assert derphi is not None, "line-search did not converge"
+
+        if derphi is None:
+            print("line-search did not converge")
```

Comment:
Can you turn the prints into logs?

I.e. at the beginning of the code do a:
```python
import logging
_log = logging.getLogger(__name__)
```

and then use `_log.info`, `_log.warning` and `_log.error` depending on what you are reporting.